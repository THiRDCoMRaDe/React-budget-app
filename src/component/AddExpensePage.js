import React from "react";
import { connect } from "react-redux";
//import {addExpense} from "../actions/expenses";
import {startAddExpense } from "../actions/expenses";
import ExpenseForm from './ExpenseForm';

export class AddExpensePage extends React.Component{
    onSubmit = (expense) => {
        this.props.startAddExpense(expense);
        this.props.history.push('/');
    };
    render() {
      return(
          <div>
              <div className="page-header">
                  <div className="content_container">
                    <h1 className="page-header__title">Add Expense</h1>
                  </div>
              </div>
              <div className="content_container">
                  <ExpenseForm
                      onsubmit={this.onSubmit}
                  />
              </div>
              {/*this is from my add expense component*/}
          </div>
      )
    }
}
const mapDispatchToProps = (dispatch) => ({startAddExpense: (expense) => dispatch(startAddExpense(expense)) });
export default connect(undefined, mapDispatchToProps)(AddExpensePage);