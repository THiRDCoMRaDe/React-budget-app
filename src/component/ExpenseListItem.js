import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({description, amount, createdAt, id}) => (
    <>
        <Link to={`/edit/${id}`}>
            <h3>
                description: {description}
            </h3>
        </Link>
        <p>
            {amount} - {createdAt}
        </p>

    </>
);


export default ExpenseListItem;
//export default ExpenseListItem;