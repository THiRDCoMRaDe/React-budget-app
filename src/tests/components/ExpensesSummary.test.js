import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from "../../component/ExpensesSummary";
import {expect} from "@jest/globals";

test('should correctly render expensesSummary with 1 expense',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});
test('should correctly render expensesSummary with multiple expenses',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={23512340987} />);
    expect(wrapper).toMatchSnapshot();
});