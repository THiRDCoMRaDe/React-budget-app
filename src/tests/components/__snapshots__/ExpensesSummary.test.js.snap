// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expensesSummary with 1 expense 1`] = `
<Fragment>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</Fragment>
`;

exports[`should correctly render expensesSummary with multiple expenses 1`] = `
<Fragment>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $235,123,409.87
  </h1>
</Fragment>
`;
